lib/semian/rails.rb

```diff
@@ -2,6 +2,8 @@
 
 class ActiveRecord::ConnectionAdapters::AbstractAdapter
   def semian_resource
-    @connection.semian_resource
+    # support for https://github.com/rails/rails/commit/d86fd6415c0dfce6fadb77e74696cf728e5eb76b
+    connection = instance_variable_defined?("@raw_connection") ? @raw_connection : @connection
```

Comment:
I don't have an opinion, in fact I'm simply curious:

I wonder if explicitly using the Rails version to discriminate on the ivar has advantages? It certainly makes the intent more clear, in that it's for backwards compatibility. I suppose it's more fragile, though.

Curious what others think.

Reply:
I feel like that question comes often and I haven't heard strong preference on this. I believe last time we were discussing something similar we agreed that using explicit Rails version branching more often used but no obvious advantage/disadvantage 

Also curious about other opinions!

Reply:
When it's not too complicated I prefer feature checking, especially when it's for yet unreleased versions, because I've seen changes like this be reverted in the past, so feature testing is more reliable.